### DISCUSSION TITLE
How could this be used in github actions for a test runner?

### DISCUSSION THREAD

**Discussion by rreynier** [gotenberg/gotenberg#704] [Q&A]
Part of my test suite is to execute code which will generate a PDF. Though I realize I could mock the api call to generate the pdf, I prefer to also have an integration test. That said, how could I get an instance of Gotenberg running inside github actions workflow? Is this possible?

**Comment by gulien** [comment]
Hello @rreynier,

A solution could be to use a `docker-compose.yml` file with your application + Gotenberg containers.

With a command like `docker compose run yourapp yourtestcmd`, it will start both your application and Gotenberg instances plus run your tests.

**Reply by rreynier** [reply]
I am already using docker-compose for local development. I use it to run all my services like mysql, redis.. and now gotenberg. Both mysql and redis are actually provided by the ubuntu image I use  in github actions (ubuntu 22). This allows my php tests to execute integration tests that talk to the db and the cache within the workflow (part of my ci). So now I need to figure out a way to support gotenberg. I could spin up a gotenberg instance external and make it publicly available, but that seems to not be recommended by your docs.
